proposals/features-composite-dependencies.md

```diff
@@ -0,0 +1,117 @@
+            "version": "1.2.3",  // An exact version is required.  We do not permit pinning to a major or minor version.
+            "detect": "a --version && cat /etc/a/.markerfile", // Only continue installation of this Feature if detect returns non-zero
+            "options": {
+                "bar": true
+            }
+        },
+        {
+            "id": "ghcr.io/microsoft/features/b",
+            "version": "sha256:45b23dee08af5e43a7fea6c4cf9c25ccf269ee113168c19722f87876677c5cb2", // SHA of the published artifact is OK
+            "detect": undefined, // Omit or set as 'undefined' to always install this Feature.
+            "options": {
+                "zip": "zap"
+            }
+        }
+    ]
+}
+```
+
+An optional `finalize.sh` script can be included, and will be fired after all Features have been installed.
+
+A composite Feature will be published following the same process as an [standalone dev container Feature](https://containers.dev/implementors/features) into the same namespace - following the pattern outlined in [the Features distribution spec](https://containers.dev/implementors/features-distribution/). Dependencies of a composite Feature can be published to the same or different namespaces.
+
+An example repository structure for a repo with one composite Feature and an standalone Feature can be found below:
+
+```
+$ tree 
+
+├── src
+│   ├── composite
+│   │   ├── README.md
+│   │   ├── devcontainer-feature.composite.json
+│   │   ├── finalize.sh
+│   ├── a
+│   │   ├── README.md
+│   │   ├── devcontainer-feature.json
+│   │   └── install.sh
+...
+```
+
+### Notes:
+
+- Composite Features cannot depend on other composite Features. This is to prevent a circular dependencies and deep dependency chains.
+- The `dependsOn` array is in the order that the Features should be installed.  This is important for Features that depend on the existence of a file or directory created by a previous Feature.
+- The `detect` property is optional.  If omitted, the Feature will always be installed.
+- The `options` property is optional.  If omitted, the default options will be passed to the Feature's `install.sh` script, as defined in the Feature's `devcontainer-feature.json` file.
+- A composite Feature can optionally include a `finalize.sh` script.  This script will be executed after all of the dependencies have been installed.  This is useful for Features that need to perform some action after all of the dependencies have been installed.
+
+
+## Advantages
+
+- Composite Features can be used to distribute a single Feature that depends on other Features.
+- Composite Features can pin all of their dependencies to a specific version, ensuring that the Feature can be tested and will work as expected.
+- Composite Features prevent the complexity that arises with deeply nested dependencies or circular dependencies.
+
+## Disadvantages
+
+- Composite features are not as flexible as other possible dependency models.
```

Comment:
Do you have an example in mind of how other models are more flexible, so that readers can more fully understand which limitation(s) this refers to?